tests/Unit/Evolution/Actions/Test_AddMeshVelocityNonconservative.cpp

```diff
@@ -0,0 +1,159 @@
+
+#include <boost/optional.hpp>
+#include <cstddef>
+#include <utility>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/TagsTimeDependent.hpp"
+#include "Evolution/Actions/AddMeshVelocityNonconservative.hpp"
+#include "Framework/ActionTesting.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+#include "Parallel/PhaseDependentActionList.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Literals.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+struct Var1 : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+
+template <size_t Dim>
+struct Var2 : db::SimpleTag {
+  using type = tnsr::I<DataVector, Dim, Frame::Inertial>;
+};
+
+template <size_t Dim, typename GradientsTags>
+struct System {
+  static constexpr size_t volume_dim = Dim;
+  using variables_tag = Tags::Variables<tmpl::list<Var1, Var2<Dim>>>;
+  using gradients_tags = GradientsTags;
+};
+
+template <typename Metavariables>
+struct component {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = int;
+  using simple_tags = tmpl::list<
+      db::add_tag_prefix<
+          ::Tags::deriv, typename Metavariables::system::variables_tag,
+          tmpl::size_t<Metavariables::volume_dim>, Frame::Inertial>,
+      db::add_tag_prefix<Tags::dt,
+                         typename Metavariables::system::variables_tag>,
+      domain::Tags::MeshVelocity<Metavariables::volume_dim>>;
+  using phase_dependent_action_list = tmpl::list<
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Initialization,
+          tmpl::list<ActionTesting::InitializeDataBox<simple_tags>>>,
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Testing,
+          tmpl::list<evolution::Actions::AddMeshVelocityNonconservative>>>;
+};
+
+template <typename System>
+struct Metavariables {
+  static constexpr size_t volume_dim = System::volume_dim;
+  using component_list = tmpl::list<component<Metavariables>>;
+  using system = System;
+  enum class Phase { Initialization, Testing, Exit };
+};
+
+template <bool HasMeshVelocity, size_t Dim, typename GradientsTags>
+void test() {
+  using system = System<Dim, GradientsTags>;
+  using metavars = Metavariables<system>;
+  using MockRuntimeSystem = ActionTesting::MockRuntimeSystem<metavars>;
+
+  boost::optional<tnsr::I<DataVector, Dim, Frame::Inertial>> frame_velocity{};
+  if (HasMeshVelocity) {
+    frame_velocity = tnsr::I<DataVector, Dim, Frame::Inertial>{};
+    for (size_t d = 0; d < Dim; ++d) {
+      frame_velocity->get(d) = DataVector{(1. + d) * 5., (1. + d) * 6.};
```

Comment:
does clang-tidy object if you put an additional layer of `{}`?

Reply:
I'm calling the constructor so `{}` is the correct syntax. Because of (what I view as) a clang bug, `std::array` (and only std::array) wants `{{}}`. It might be that newer versions of clang are now okay with `{}` for arrays. The reason `tensor` gets the horrid `{{{}}}` treatment is because this invokes the `Tensor(std::array)` constructor, and the `array` needs `{{}}` while `Tensor` needs `{}`. So the answer to your question is: I'm not sure, but it also is not correct C++ so I doubt it would compile